### PR TITLE
Fix  `parentId` when choosing `auto-increment` id type

### PR DESCRIPTION
**Pull request by abrl91**
Close: #4677 

## PR Details
Fix  `parentId` when choosing `auto-increment` id type

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

### CODE REVIEW

File: packages/amplication-data-service-generator/src/server/resource/service/create-service.ts [hunk 1]
```diff
@@ -346,6 +357,7 @@ async function createToOneRelationFile(
 
   interpolate(toOneFile, {
     DELEGATE: delegateId,
+    PARENT_ID_TYPE: identityParentIdType(entity),
```
Comment (yuval-hazaz) [inline packages/amplication-data-service-generator/src/server/resource/service/create-service.ts:360]: How do you know which field in the entity to look at just by sending the entity to the function without the current field?

File: packages/amplication-data-service-generator/src/server/resource/service/create-service.ts [hunk 2]
```diff
@@ -375,6 +389,23 @@ async function createToManyRelationFile(
   return toManyFile;
 }
 
+function identityParentIdType(entity: Entity): namedTypes.Identifier {
+  const relatedEntityFiledId = entity.fields?.find(
+    (relatedEntityField) => relatedEntityField.dataType === EnumDataType.Id
```
Comment (yuval-hazaz) [inline packages/amplication-data-service-generator/src/server/resource/service/create-service.ts:394]: By doing this, you will always find the ID field of the entity and not the ID field of the related entity.

File: packages/amplication-data-service-generator/src/server/resource/service/create-service.ts [hunk 3]
```diff
@@ -375,6 +389,23 @@ async function createToManyRelationFile(
...
+  const { idType } = (relatedEntityFiledId?.properties as types.Id) || "CUID";
+  const idTypeTSOptions: {
+    [key in types.Id["idType"]]: namedTypes.Identifier;
+  } = {
+    AUTO_INCREMENT: builders.identifier("number"),
+    UUID: builders.identifier("string"),
+    CUID: builders.identifier("string"),
+  };
+
+  return idTypeTSOptions[idType] ?? idTypeTSOptions["CUID"];
```
Comment (yuval-hazaz) [inline packages/amplication-data-service-generator/src/server/resource/service/create-service.ts:406]: what is this for (idTypeTSOptions["CUID"]) ?

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

File: packages/amplication-data-service-generator/src/server/resource/service/create-service.ts (hunk 2 above)
Comment (abrl91) [reply packages/amplication-data-service-generator/src/server/resource/service/create-service.ts:394]: You're right. I started with the related entity, but I tested it and it didn't work. I got the same error.

Comment (abrl91) [reply packages/amplication-data-service-generator/src/server/resource/service/create-service.ts:394]: I forgot to rename the variables

File: packages/amplication-data-service-generator/src/server/resource/service/create-service.ts (hunk 1 above)
Comment (abrl91) [reply packages/amplication-data-service-generator/src/server/resource/service/create-service.ts:360]: The function is searching for the id type
```
const entityFiledId = entity.fields?.find(
    (entityField) => entityField.dataType === EnumDataType.Id
  );
```

File: packages/amplication-data-service-generator/src/server/resource/service/create-service.ts (hunk 3 above)
Comment (abrl91) [reply packages/amplication-data-service-generator/src/server/resource/service/create-service.ts]: for the default

Comment (abrl91) [reply packages/amplication-data-service-generator/src/server/resource/service/create-service.ts]: I logged the idType and for the entities that don't have this property, it returns `undefined`
image

File: packages/amplication-data-service-generator/src/util/get-entity-id-type.ts [hunk 4]
```diff
@@ -3,17 +3,17 @@ import DsgContext from "../dsg-context";
...
+export const getEntityIdType = (entityName: string) => {
   const { entities } = DsgContext.getInstance;
-  const userEntity = entities.find(
-    (entity) => entity.name === USER_ENTITY_NAME
-  );
+  const entity = entities.find((entity) => entity.name === entityName);
 
-  if (!userEntity) {
+  if (!entity) {
     throw new Error("User entity not found");
```
Comment (yuval-hazaz) [inline packages/amplication-data-service-generator/src/util/get-entity-id-type.ts]: update message (remove "user" or use dynamic value with ${entityName}

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**